homeassistant/components/dyson/fan.py

```diff
@@ -171,22 +200,69 @@ def service_handle(service):
             DYSON_DOMAIN, SERVICE_SET_TIMER, service_handle, schema=SET_TIMER_SCHEMA
         )
 
-        hass.services.register(
-            DYSON_DOMAIN,
-            SERVICE_SET_DYSON_SPEED,
-            service_handle,
-            schema=SET_DYSON_SPEED_SCHEMA,
-        )
-
 
 class DysonFanEntity(DysonEntity, FanEntity):
     """Representation of a Dyson fan."""
 
+    @property
+    def speed(self):
+        """Return the current speed."""
+        return SPEED_DYSON_TO_HA[self._device.state.speed]
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
+
+    @property
+    def dyson_speed(self):
+        """Return the current speed."""
+        if self._device.state:
+            if self._device.state.speed == FanSpeed.FAN_SPEED_AUTO.value:
+                return self._device.state.speed
+            return int(self._device.state.speed)
```

Comment:
Please return `None` explicitly here.
```suggestion
            return int(self._device.state.speed)
        return None
```

Reply:
The state check is unnecessary. Removed.